### PR TITLE
Add Google Signin button to login page

### LINKED ISSUE #6713: Add Google Sign in support
Use an iframe based approach to allow login into open library using a google account, similar to archive.org

### PR DESCRIPTION
**Pull request by cdrini**
Closes #6713

### CODE REVIEW

File: openlibrary/templates/account/ia_thirdparty_logins.html [hunk 1]
```diff
@@ -0,0 +1,31 @@
+$def with ()
+
+$ params = {'origin': request.home.replace('http:', 'https:')}
+<iframe
+    id="ia-third-party-logins"
+    src="https://www-drini.archive.org/account/login.thirdparty.php?$(urlencode(params))"
```
Comment (mekarpeles) [inline openlibrary/templates/account/ia_thirdparty_logins.html:6]:
```suggestion
    src="https://archive.org/account/login.thirdparty.php?$(urlencode(params))"

```

**Review by mekarpeles — APPROVED** [review]
lgtm other than updating urls to prod

**Comment by cdrini** [comment]
Blocked waiting for deploy to archive.org production of sibling MR on that side.

File: openlibrary/accounts/model.py [hunk 2]
```diff
@@ -848,13 +852,13 @@ def audit_accounts(
         # lending.config_ia_auth_only is enabled, we need to create
         # and link it.
         if not ol_account:
-            if not password:
-                raise {'error': 'link_attempt_requires_password'}
             try:
                 ol_account = OpenLibraryAccount.create(
```
Comment (mekarpeles) [inline openlibrary/accounts/model.py:856]:
```suggestion
                # Db update will fail in infogami if web.ctx['ip'] not set
                ol_account = OpenLibraryAccount.create(
```

Comment (cdrini) [reply openlibrary/accounts/model.py:856]: Mek opted out of this one

File: static/css/base/helpers-misc.less [hunk 3]
```diff
@@ -128,6 +128,9 @@ button {
 .neg,
 .red {
   color: @red !important;
+  &:empty {
+    display: none;
+  }
```
Comment (jimchamp) [inline static/css/base/helpers-misc.less:133]: Is there a better way to do this?  I believe that this is the root cause of #6933

Comment (cdrini) [reply static/css/base/helpers-misc.less:133]: Fixed; good find on the root cause!

**Pull request merged**